### PR TITLE
TRINITY-54: Create api endpoints for Assignments and AssignmentSubmissions

### PR DESCRIPTION
**Pull request by stetsche**
Added api endpoints for Assignments and AssignmentSubmissions

Removed some unused constants in AssignmentConstants

### CODE REVIEW

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/AssignmentsController.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            AssignmentRestBean bean = new AssignmentRestBean(assignment);
+            bean.setStatus(assignmentService.getAssignmentStatus(assignment.getId()));
+            bean.setScale(assignmentService.getScaleDisplay(assignment.getId()));
+            bean.setAccess(assignmentService.getAccessDisplay(assignment.getId()));
+            String assignmentReference = assignmentService.assignmentReference(siteId, assignment.getId());
+            bean.setTotalSubmissions(assignmentService.countSubmissions(assignmentReference, null));
+            bean.setNewSubmissions(assignmentService.countSubmissions(assignmentReference, false));
+            return bean;
+        }).collect(Collectors.toList());
+        return beanList;
```
Comment (adrianfish) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/AssignmentsController.java:71]:
```suggestion
        return assignmentService.getAssignmentsForContext(siteId).stream().map(assignment -> {
            AssignmentRestBean bean = new AssignmentRestBean(assignment);
            bean.setStatus(assignmentService.getAssignmentStatus(assignment.getId()));
            bean.setScale(assignmentService.getScaleDisplay(assignment.getId()));
            bean.setAccess(assignmentService.getAccessDisplay(assignment.getId()));
            String assignmentReference = assignmentService.assignmentReference(siteId, assignment.getId());
            bean.setTotalSubmissions(assignmentService.countSubmissions(assignmentReference, null));
            bean.setNewSubmissions(assignmentService.countSubmissions(assignmentReference, false));
            return bean;
        }).collect(Collectors.toList());
```

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/AssignmentsController.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    @GetMapping(value = "/assignments/{assignmentId}/submissions", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (adrianfish) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/AssignmentsController.java:74]: I think you should stick to /sites/{siteId}/assignments/{assignmentId}/submissions. I know it's a bit longer, but it's consistent with other parts of the webapi. But, I'm open to discussion.

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/AssignmentsController.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    public List<AssignmentSubmissionRestBean> getSiteAssignmentSubmissions(@PathVariable String assignmentId) {
+
+        String currentUserId = checkSakaiSession().getUserId();
+
+        try {
+            Assignment assignment = assignmentService.getAssignment(assignmentId);
+
+            //Do not return Submissions for deleted Assignments
+            if (assignment == null | assignment.getDeleted()) {
```
Comment (adrianfish) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/AssignmentsController.java:83]: Binary OR ??

Comment (adrianfish) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/AssignmentsController.java:83]: I don't think you intended that!

Comment (adrianfish) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/AssignmentsController.java:83]: although, same result

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/AssignmentsController.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            //Do not return Submissions for deleted Assignments
+            if (assignment == null | assignment.getDeleted()) { 
+                log.warn("User {} requested submissions for non existing or (soft)deleted Assingment {}", currentUserId, assignmentId);
+                return Collections.emptyList(); 
+            }
+
+            List<String> userIds = assignmentService.getSubmitterIdList(null, "all", null, 
+                assignmentService.assignmentReference(assignment.getContext(), assignment.getId()), assignment.getContext());
+
+            List<AssignmentSubmissionRestBean> beanList = userDirectoryService.getUsers(userIds).stream().map(user -> {
```
Comment (adrianfish) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/AssignmentsController.java:91]: just return the list straight off, no variable needed

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/AssignmentsController.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ */
+@Slf4j
+@RestController
+public class AssignmentsController extends AbstractSakaiApiController {
+
+	@Autowired
```
Comment (adrianfish) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/AssignmentsController.java:45]: tabbing

**Review by adrianfish — CHANGES_REQUESTED** [review]
Some small stuff

**Comment by profmikegreene** [comment]
I didn't get a chance to look at this yesterday, does it need a rebase or something? Is it supposed to be this many commits/files @stetsche

**Comment by stetsche** [comment]
thanks, fixed it

**Comment by mpellicer** [comment]
Is the Java build broken?

**Comment by stetsche** [comment]
the build here is failing in library, but yes actually there is a problem in webapi too

**Comment by stetsche** [comment]
> Is the Java build broken?
Assignments and webapi are building fine. It is failing in library.

File: webapi/src/main/java/org/sakaiproject/webapi/beans/AssignmentRestBean.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    private String access;
+    private String scale;
+    private Long release;
+    private Long retract;
+    private int totalSubmissions;
+    private int newSubmissions;
+    private boolean hasAttachment;
+
+    public AssignmentRestBean(Assignment assignment) {
+        id = assignment.getId();
```
Comment (ern) [inline webapi/src/main/java/org/sakaiproject/webapi/beans/AssignmentRestBean.java:37]: Would add a Objects.requireNonNull assignment check

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/AssignmentsController.java [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    AssignmentService assignmentService;
+
+    @Autowired
+    EntityManager entityManager;
+
+    @Autowired
+    UserDirectoryService userDirectoryService;
+
+    @GetMapping(value = "/sites/{siteId}/assignments", produces = MediaType.APPLICATION_JSON_VALUE)
+    public List<AssignmentRestBean> getSiteAssignments(@PathVariable String siteId) {
```
Comment (ern) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/AssignmentsController.java:56]: return a Collection vs List

File: webapi/src/main/java/org/sakaiproject/webapi/beans/AssignmentSubmissionRestBean.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+ ******************************************************************************/
+package org.sakaiproject.webapi.beans;
+
+import org.sakaiproject.assignment.api.model.Assignment;
+import org.sakaiproject.assignment.api.model.AssignmentSubmission;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
```
Comment (ern) [inline webapi/src/main/java/org/sakaiproject/webapi/beans/AssignmentSubmissionRestBean.java:22]: Maybe use @Data or @EqualsAndHashcode and set the field to be the id for the equality check

File: webapi/src/main/java/org/sakaiproject/webapi/beans/AssignmentSubmissionRestBean.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    private String title;
+    private String grade;
+    private String status;
+    private String statusText;
+    private boolean released;
+    private boolean hasAttachment;
+    private Long submittedDate;
+
+    public AssignmentSubmissionRestBean(AssignmentSubmission assignmentSubmission) {
+        Assignment assignment = assignmentSubmission.getAssignment();
```
Comment (ern) [inline webapi/src/main/java/org/sakaiproject/webapi/beans/AssignmentSubmissionRestBean.java:37]: Add a check for a null assignmentSubmission and check for a empty id (maybe StringUtils.isBlank)

**Review by profmikegreene — COMMENTED** [review]
Looks good for what trinity project needs. Thanks!

**Review by mpellicer — APPROVED** [review]

**Pull request merged**